### ISSUE TITLE
False Positive: `_modify` for analyzer rules

### ISSUE THREAD

**Issue by Iron-Ham** [realm/SwiftLint#5094]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When using the `unused_declarations` analyze rule, I noticed that `_modify` blocks are erroneously triggering the analyzer; that would be akin to a `get` or `set` block triggering the analyzer.

Note that when we scan it for errors, declarations like this will produce an error on the `_modify` keyword:

```swift

  @inlinable public subscript<T: AnyScalarType & Hashable>(_ key: String) -> T {
    get {
#if swift(>=5.4)
        _data[key] as! T
#else
        _data[key]?.base as! T
#endif
    }
    set { _data[key] = newValue }
    _modify {
      var value = _data[key] as! T
      defer { _data[key] = value }
      yield &value
    }
  }
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ ➜ swiftlint analyze --compiler-log-path xcodebuild.log
```

It's quite long, so I'll just filter it for the errors:

```
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/GraphQLHTTPMethod.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/Collection+Helpers.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/PossiblyDeferred.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/NormalizedCache.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/SelectionSet+JSONInitializer.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/GraphQLSelectionSetMapper.swift:5:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/RequestChain.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/RequestBodyCreator.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/InMemoryNormalizedCache.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/DataLoader.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/HTTPURLResponse+Helpers.swift:8:7: error: Unused Declaration Violation: Declarations should be referenced at least once within all files linted (unused_declaration)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/DispatchQueue+Optional.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/Apollo/GraphQLResult.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/InputValue.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/LocalCacheMutation.swift:39:5: error: Unused Declaration Violation: Declarations should be referenced at least once within all files linted (unused_declaration)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/LocalCacheMutation.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/DataDict.swift:35:5: error: Unused Declaration Violation: Declarations should be referenced at least once within all files linted (unused_declaration)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/DataDict.swift:45:5: error: Unused Declaration Violation: Declarations should be referenced at least once within all files linted (unused_declaration)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/JSON.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/GraphQLNullable.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/AnyHashableConvertible.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/Selection.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)
/Users/zer0/Developer/OSS/apollo-ios/Sources/ApolloAPI/Selection+Conditions.swift:1:1: warning: Unused Import Violation: All imported modules should be required to make the file compile (unused_import)

````

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.52.3
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Brew
* Paste your configuration file:

```yml
analyzer_rules:
  - unused_import
  - unused_declaration
```

* Which Xcode version are you using (check `xcodebuild -version`)? 14.3.1

```swift
  @inlinable public subscript<T: AnyScalarType & Hashable>(_ key: String) -> T {
    get {
#if swift(>=5.4)
        _data[key] as! T
#else
        _data[key]?.base as! T
#endif
    }
    set { _data[key] = newValue }
    _modify {
      var value = _data[key] as! T
      defer { _data[key] = value }
      yield &value
    }
  }
```

**Comment by SimplyDanny** [comment]
Looks like the SourceKit version bundled with Xcode is not able to detect the `_modify` as a special kind of accessor. It's currently tagged as a method declaration. So there doesn't seem to be a proper way to recognize it. In case I'm right, I wouldn't blame SourceKit to lag support for an unofficial language feature.
